jac-uk/admin
# Feature/2230 Character declaration updates

## What's included?
Closes #2230 

- [x] Reword 'Recommendation' to: **Recommended SCC approach**.
- [x] Force Character issues from the current exercise to be visible onscreen in the default view; at the moment they are not being displayed
- [x] Reword ‘View all issues’ to **_View all character issues from previous applications_**
- [x] Fix recommendation dropdown so that selected option remains onscreen when chosen, it currently remains blank when the an option is selected.
- [x] Fix offence category dropdown so that selected option remains onscreen when chosen,  it currently remains blank when the an option is selected.
- [x] Fix search functionality of search box and add guidance on what can be searched as per other search functionality on Admin
- [x] Where no declaration provided on previous exercises due to the candidate not having been selected and the data not requested, the platform should add **Data not requested** next to each exercise that the candidate has applied for where there is no entry 

## Who should test?
✅ Product owner
✅ Developers
✅ UTG

## How to test?
- Go to the preview link: https://jac-admin-develop--pr2278-feature-2230-charact-dzyc1799.web.app/exercise/aDxEqK36807d0Nhpugo3/reports/character-issues
- Click `Refresh` to generate latest report
- `Recommended SCC Approach Field`
    - Locate the "Recommended SCC Approach" field (refer to place 1 in the image).
    - Confirm that this field is visible for each candidate.
- Selected Option in `Recommended SCC Approach Field`
    - Check the "Recommended SCC Approach" field for each candidate.
    Verify that the selected option is displayed in this field (refer to place 2 in the image).
- `View All Character Issues` Link
    - Look for the "View all character issues from previous applications" link (refer to place 3 in the image).
    - Ensure that this link is visible for each candidate.
- Selected Option in `Offence Category` Field
    - Check the "Offence Category" field for each candidate.
    - Confirm that the selected option is displayed in this field (refer to place 4 in the image).  
![image](https://github.com/jac-uk/admin/assets/79906532/81b5b2b3-c8e5-457f-b768-5a13a753ec62)
- Check the `Search application records` functionality
    - Click `Help`, it should appear searchable fields:
![image](https://github.com/jac-uk/admin/assets/79906532/3278311f-265c-4c27-8dfa-d38dbaa9d260)
    - Enter a least 3 letters, it should filter the conditates accordingly.
![image](https://github.com/jac-uk/admin/assets/79906532/8abdb8b3-2e55-4e87-9b28-3fb653e51f03)
- Go to the preview link: https://jac-admin-develop--pr2278-feature-2230-charact-dzyc1799.web.app/exercise/7LgaEMudSSfBC1fOwnaf/reports/character-issues
   - Click `Refresh` to generate latest report
    - Click `View all character issues from previous applications` of the candidtate `application-0012 Jac`
    - If the candidate not having previous applications, it should appear `No previous applications with character issues `
![image](https://github.com/jac-uk/admin/assets/79906532/28876c5f-7731-4e73-9c84-344df3f15200)
    - Click `View all character issues from previous applications` of the candidtate `application-0001 Jac`
    - If the candidate has character issues in previous exercise, it should list the issues.
    ![image](https://github.com/jac-uk/admin/assets/156695133/dbf3428f-0b3a-44ea-9e6c-6a5f1ae3ba2c)
    - If the candidate not having been selected and the data not requested, it should show `Data not requested` in exercise.
![image](https://github.com/jac-uk/admin/assets/79906532/59a0bac4-490d-4f81-8990-82b003787c8a)





    

## Risk - how likely is this to impact other areas?
🟢 No risk - this is a self-contained piece of work

## Additional context
Include screen grabs, video demo, notes etc.

## Related permissions
Have permissions been considered for this functionality?
- No permission changes required
- Permissions have been added / updated. Details:

---
PREVIEW:DEVELOP
_can be OFF, DEVELOP or STAGING_
